### PR TITLE
Change kube library to pykube-ng

### PR DESCRIPTION
**Pull request by mikkeloscar**
This changes the Kubernetes library to https://github.com/hjacobs/pykube which is a fork and updated version of the abandoned https://github.com/kelproject/pykube library.

This change is needed in order to work with Kubernetes v1.16 which doesn't support the extensions/v1beta1 APIs for deployments.

### CODE REVIEW

**Comment by hjacobs** [comment]
How can we test that everything works? It should work out-of-the-box, but I also removed some utility functions (but I don't think ZMON Worker was using them).

**Comment by mikkeloscar** [comment]
I did the same update for https://github.com/zalando-zmon/zmon-agent-core/pull/119 which I tested manually.

This is a bit harder I realize, especially because the worker needs python 2.7.

**Comment by hjacobs** [comment]
Ah, yes, ZMON Worker is still Python 2.7 :disappointed: (Python 2.7 is officially unmaintained since January 1st.. https://pythonclock.org/)

**Comment by mikkeloscar** [comment]
As suggested by @hjacobs I have instead of updating the library simply imported the relevant classes which we needed to update.

This can later be removed when zmon-worker no longer depends on python 2.7 and depend on `pykube-ng` instead.

File: zmon_worker_monitor/builtins/plugins/kubernetes.py [hunk 1]
```diff
@@ -20,6 +22,78 @@
...
+
+    @property
+    def ready(self):
+        return (
+            self.obj["status"]["observedGeneration"]
+            >= self.obj["metadata"]["generation"]
+            and self.obj["status"]["updatedReplicas"] == self.replicas
+        )
+
+    def rollout_undo(self, target_revision=None):
```
Comment (hjacobs) [inline zmon_worker_monitor/builtins/plugins/kubernetes.py]: you can remove this function as it's not used

**Comment by aermakov-zalando** [comment]
Didn't see this PR. :( We could also do [this](https://github.com/zalando-zmon/zmon-worker/pull/432), no need to copy-paste the types.

**Comment by mikkeloscar** [comment]
@aermakov-zalando all good. Closing this one. Thanks!

**Pull request closed without merging**